<launch>
  <!-- Localization manager configuration -->
  <arg name="predicted_robot_pose_topic"/>
  <arg name="localize_action"/>
  <arg name="simulation"       default="false"/>
  <arg name="cmd_vel_mux_input" default="/cmd_vel_mux/inputs/navigation"/>
  <arg name="global_frame"      default="map"/>
  <arg name="odom_frame"        default="odom"/> 
  <arg name="base_frame"        default="base_footprint"/>

  <arg name="use_map_topic"     default="true"/>
  <arg name="scan_topic"        default="scan"/>
  <arg name="initialpose_topic" default="initialpose"/>
  <arg name="map_topic"         default="map"/>
  <arg name="odom_topic"        default="odom"/>
  <arg name="initial_pose_x"    default="0.0"/>
  <arg name="initial_pose_y"    default="0.0"/>
  <arg name="initial_pose_a"    default="0.0"/>

  <node pkg="yocs_localization_manager" name="localization_manager" type="localization_manager.py">
    <param name="simulation"               value="$(arg simulation)"/>
    <param name="simulation_global_frame"  value="$(arg global_frame)"/>
    <param name="simulation_x"             value="$(arg initial_pose_x)"/>
    <param name="simulation_y"             value="$(arg initial_pose_y)"/>
    <param name="simulation_a"             value="$(arg initial_pose_a)"/>
    <remap from="pose_tracker/poses"    to="$(arg predicted_robot_pose_topic)"/>
    <remap from="localize"              to="$(arg localize_action)"/> 
    <remap from="initialpose"           to="$(arg initialpose_topic)"/>
    <remap from="cmd_vel"               to="$(arg cmd_vel_mux_input)"/>
    <remap from="odom"                  to="$(arg odom_topic)"/>
  </node>

  <group unless="$(arg simulation)">
    <include file="$(find yocs_localization_manager)/launch/includes/amcl.launch.xml">
      <arg name="global_frame"          value="$(arg global_frame)"/>
      <arg name="base_frame"            value="$(arg base_frame)"/>
      <arg name="odom_frame"            value="$(arg odom_frame)"/>
      <arg name="initialpose_topic"     value="$(arg initialpose_topic)"/>
      <arg name="scan_topic"            value="$(arg scan_topic)"/>
      <arg name="use_map_topic"         value="$(arg use_map_topic)"/>
      <arg name="map_topic"             value="$(arg map_topic)"/>
      <arg name="initial_pose_x"        value="$(arg initial_pose_x)"/>
      <arg name="initial_pose_y"        value="$(arg initial_pose_y)"/>
      <arg name="initial_pose_a"        value="$(arg initial_pose_a)"/>
    </include>
  </group>
  <group if="$(arg simulation)">
    <node pkg="fake_localization" name="fake_localization" type="fake_localization">
      <remap from="base_pose_ground_truth" to="$(arg odom_topic)"/>
      <remap from="initialpose" to="$(arg initialpose_topic)"/>
      <param name="odom_frame_id" value="$(arg odom_frame)"/>
      <param name="global_frame_id" value="$(arg global_frame)"/>
      <param name="base_frame_id" value="$(arg base_frame)"/>
      <param name="delta_x" value="-$(arg initial_pose_x)"/>
      <param name="delta_y" value="-$(arg initial_pose_y)"/>
      <param name="delta_yaw" value="$(arg initial_pose_a)"/>
    </node>
  </group>
</launch>
